codec_record.go

```diff
@@ -327,7 +333,8 @@ type recordMapEncoder struct {
 
 func (e *recordMapEncoder) Encode(ptr unsafe.Pointer, w *Writer) {
 	for _, field := range e.fields {
-		valPtr := e.mapType.UnsafeGetIndex(ptr, reflect2.PtrOf(field))
+		key := field.name
+		valPtr := e.mapType.UnsafeGetIndex(ptr, reflect2.PtrOf(&key))
```

Comment:
This is not really doing anything. The reason is the way Go lays structs out in memory. When you have an unsafe pointer to a struct, you actually have a pointer to the first field in the struct, in this case `name`.

Reply:
You're right! It took me a while to figure out this behavior. I decided to add  ```key``` variable, regardless, to make the code clearer (IMO) and add some robustness; as you mentioned, the current implementation might break if ```mapEncoderField``` changes its first field.

Reply:
I'll undo this change